dom.go

```diff
@@ -1155,45 +1157,102 @@ func requestAnimationFrame(callback func(float64)) int {
 }
 
 // RenderBody renders the given component as the document body. The given
-// Component's Render method must return a "body" element.
+// Component's Render method must return a "body" element or a panic will
+// occur.
+//
+// This function blocks forever in order to prevent the program from exiting,
+// which would prevent components from rerendering themselves in the future.
+//
+// It is a short-handed form for writing:
+//
+// 	err := vecty.RenderInto("body", body)
+// 	if err !== nil {
+// 		panic(err)
+// 	}
+// 	select{} // run Go forever
+//
 func RenderBody(body Component) {
+	target := global.Get("document").Call("querySelector", "body")
+	err := renderIntoNode("RenderBody", target, body)
+	if err != nil {
+		panic(err)
+	}
+	if !isTest {
+		select {} // run Go forever
+	}
+}
+
+// ElementMismatchError is returned when the element returned by a component
+// does not match what is required for rendering.
+type ElementMismatchError struct {
+	method, got, want string
+}
+
+func (e ElementMismatchError) Error() string {
+	return "vecty: " + e.method + `: expected Component.Render to return a "` + e.want + `", found "` + e.got + `"`
+}
+
+// InvalidTargetError is returned when the element targeted by a render is
+// invalid because it is null or undefined.
+type InvalidTargetError struct {
+	method string
+}
+
+func (e InvalidTargetError) Error() string {
+	return "vecty: " + e.method + `: invalid target element is null or undefined`
+}
+
+// RenderInto renders the given component into the existing HTML element found
+// by the CSS selector (e.g. "#id", ".class-name") by replacing it.
+//
+// If there is more than one element found, the first is used. If no element is
+// found, an error of type InvalidTargetError is returned.
+//
+// If the Component's Render method does not return an element of the same type,
+// an error of type ElementMismatchError is returned.
+func RenderInto(selector string, c Component) error {
+	target := global.Get("document").Call("querySelector", selector)
+	return renderIntoNode("RenderInto", target, c)
+}
+
+func renderIntoNode(methodName string, node jsObject, c Component) error {
+	if !node.Truthy() {
+		return InvalidTargetError{method: methodName}
+	}
 	// block batch until we're done
 	batch.scheduled = true
-	nextRender, skip, pendingMounts := renderComponent(body, nil)
+	nextRender, skip, pendingMounts := renderComponent(c, nil)
 	if skip {
-		panic("vecty: RenderBody Component.SkipRender illegally returned true")
+		panic("vecty: " + methodName + ": Component.SkipRender illegally returned true")
 	}
-	if nextRender.tag != "body" {
-		panic("vecty: RenderBody expected Component.Render to return a body tag, found \"" + nextRender.tag + "\"")
+	expectTag := node.Get("nodeName").String()
```

Comment:
I believe I'm running into an issue with `nodeName` case sensitivity here (see https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeName and https://johnresig.com/blog/nodename-case-sensitivity/).

My program is doing:

```Go
// Initial frontend render.
err := vecty.RenderInto("body", body)
if err != nil {
	panic(fmt.Errorf("internal error: unexpected error from vecty.RenderInto: %v", err))
}
```

And getting:

```
Uncaught Error: internal error: unexpected error from vecty.RenderInto: vecty: RenderInto: expected Component.Render to return a "BODY", found "body"
```

I'm using HTML, Chrome 78.0.3904.108 on macOS.

Reply:
I tried the following quick change to see if it would be sufficient to solve the problem:

```diff
diff --git a/dom.go b/dom.go
index 0eb000e..164a5a3 100644
--- a/dom.go
+++ b/dom.go
@@ -2,6 +2,7 @@ package vecty
 
 import (
 	"reflect"
+	"strings"
 )
 
 // batch renderer singleton
@@ -1225,7 +1226,7 @@ func renderIntoNode(methodName string, node jsObject, c Component) error {
 	if skip {
 		panic("vecty: " + methodName + ": Component.SkipRender illegally returned true")
 	}
-	expectTag := node.Get("nodeName").String()
+	expectTag := strings.ToLower(node.Get("nodeName").String())
 	if nextRender.tag != expectTag {
 		return ElementMismatchError{method: methodName, got: nextRender.tag, want: expectTag}
 	}
```

It was; after that, Go Package Store was rendering the frontend without issues.

I'm not sure if that's the optimal long term fix, or if there's a better way. You may not want to import a new `strings` package just for one function.

I tried using JavaScript's `toLowerCase` method, but for some reason that didn't work in the context of Vecty:

```
Uncaught Error: syscall/js: call of Value.Call on string
```

Even though it seems to work fine elsewhere, e.g., see https://gopherjs.github.io/playground/#/UxD9vDjbf0.

Hope this is helpful.

Reply:
About the "syscall/js: call of Value.Call on string" error, I found it happens when using `syscall/js` package with GopherJS but not when using `github.com/gopherjs/gopherjs/js` package. So it's not Vecty-specific beyond Vecty using `syscall/js`.

It happens when using `syscall/js` with both GopherJS and Go/WebAssembly, so I filed golang/go#35917 to learn more about that.

Reply:
Thank you for investigating this! I've sent #251 to fix the issue on the Vecty side